rotkehlchen/chain/ethereum/manager.py

```diff
@@ -974,3 +976,36 @@ def get_blocknumber_by_time(self, ts: Timestamp, etherscan: bool = True) -> int:
             except RemoteError:
                 pass
         return self._get_blocknumber_by_time_from_subgraph(ts)
+
+    def get_basic_contract_info(self, address: ChecksumEthAddress) -> Dict[str, Any]:
+        """
+        Query a contract address and return basic information as:
+        - Decimals
+        - name
+        - symbol
+        if it is provided in the contract. This method may raise:
+        - BadFunctionCallOutput: If there is an error calling a bad address
+        """
+        properties = ('decimals', 'symbol', 'name')
+        info: Dict[str, Any] = {}
+
+        contract = EthereumContract(address=address, abi=ERC20TOKEN_ABI, deployed_block=0)
+        try:
+            # Output contains call status and result
+            output = multicall_2(
+                ethereum=self,
+                require_success=False,
+                calls=[(address, contract.encode(method_name=prop)) for prop in properties],
+            )
+        except RemoteError:
+            pass
+
+        decoded = [
+            contract.decode(x[1], method_name)[0] if x[0] else None  # pylint: disable=E1136
+            for (x, method_name) in zip(output, properties)
+        ]
```

Comment:
There is an error here. I just tried to use it with a random (non contract) address and it did not work properly.

```
http GET http://localhost:5042/api/1/blockchains/ETH/erc20details/ address=0x35bD01FC9d6D5D81CA9E055Db88Dc49aa1c699A8
```

I get an exception

```
Traceback (most recent call last):
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/flask_restful/__init__.py", line 468, in wrapper
    resp = resource(*args, **kwargs)
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/flask/views.py", line 89, in view
    return self.dispatch_request(*args, **kwargs)
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/flask_restful/__init__.py", line 583, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/webargs/core.py", line 366, in wrapper
    return func(*args, **kwargs)
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/api/v1/resources.py", line 1500, in get
    return self.rest_api.get_token_information(address, async_query)
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/api/rest.py", line 168, in wrapper
    return f(wrappingobj, *args, **kwargs)
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/api/rest.py", line 2834, in get_token_information
    response = self._get_token_info(token_address)
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/api/rest.py", line 2816, in _get_token_info
    info = eth_manager.get_basic_contract_info(address=address)
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/chain/ethereum/manager.py", line 1005, in get_basic_contract_info
    for (x, method_name) in zip(output, properties)
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/chain/ethereum/manager.py", line 1005, in <listcomp>
    for (x, method_name) in zip(output, properties)
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/chain/ethereum/contracts.py", line 71, in decode
    return WEB3.codec.decode_abi(output_types, result)
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/eth_abi/codec.py", line 181, in decode_abi
    return decoder(stream)
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/eth_abi/decoding.py", line 127, in __call__
    return self.decode(stream)
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/eth_utils/functional.py", line 45, in inner
    return callback(fn(*args, **kwargs))
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/eth_abi/decoding.py", line 173, in decode
    yield decoder(stream)
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/eth_abi/decoding.py", line 127, in __call__
    return self.decode(stream)
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/eth_abi/decoding.py", line 198, in decode
    raw_data = self.read_data_from_stream(stream)
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/eth_abi/decoding.py", line 308, in read_data_from_stream
```

This is because of the way this multicall contract responds if the function is missing. It does not return `None`. It returns `b''`.

It's easy to see if you go in with a debugger right before the exception:

```
(Pdb++) pp output
[(True, b''), (True, b''), (True, b'')]
```